pkg/kubectl/cmd/label.go

```diff
@@ -254,11 +254,22 @@ func (o *LabelOptions) RunLabel() error {
 		var outputObj runtime.Object
 		dataChangeMsg := "not labeled"
 		if o.dryrun || o.local || o.list {
-			err = labelFunc(info.Object, o.overwrite, o.resourceVersion, o.newLabels, o.removeLabels)
+			obj := info.Object
+			oldObj, err := json.Marshal(obj)
 			if err != nil {
 				return err
 			}
-			dataChangeMsg = "labeled"
+			err = labelFunc(obj, o.overwrite, o.resourceVersion, o.newLabels, o.removeLabels)
+			if err != nil {
+				return err
+			}
+			newObj, err := json.Marshal(obj)
+			if err != nil {
+				return err
+			}
+			if !reflect.DeepEqual(oldObj, newObj) {
+				dataChangeMsg = "labeled"
+			}
```

Comment:
You're duplicating code, can you make a helper that will return the `dataChangeMsg` and hide there all the rest?

Reply:
@soltysh Modified, PTAL